dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java

```diff
@@ -0,0 +1,225 @@
+    final int offset, maxResults;
+    final boolean showWorking, showArchived, showFolders, showFiles, showPages,sortByDesc, showLinks, showDotAssets;
+    final long languageId;
+    final List<String> extensions, mimeTypes;
+    
+    @Override
+    public String toString() {
+        return "BrowserQuery {user:" + user + ", hostFolderId:" + hostFolderId + ", filter:" + filter + ", sortBy:" + sortBy + ", offset:" + offset + ", maxResults:" + maxResults + ", showWorking:" + showWorking
+                        + ", showArchived:" + showArchived + ", showFolders:" + showFolders + ", onlyFiles:" + showFiles
+                        + ", sortByDesc:" + sortByDesc + ", showLinks:" + showLinks + ", showDotAssets:" + showDotAssets + ", showPages:" + showPages+ ", languageId:"
+                        + languageId + ", extensions:" + extensions + ", mimeTypes:" + mimeTypes + "}";
+    }
+
+
+    private BrowserQuery(Builder builder) {
+        this.user = (builder.user==null) ? APILocator.systemUser() : builder.user;
+        this.hostFolderId = builder.hostFolderId;
+        this.filter = builder.filter;
+        this.mimeTypes = builder.mimeTypes;
+        this.extensions = builder.extensions;
+        this.sortBy = UtilMethods.isEmpty(builder.sortBy) ? "moddate" : builder.sortBy;
+        this.offset = builder.offset;
+        this.maxResults = builder.maxResults>500 ? 500 : builder.maxResults;
+        this.showWorking = builder.showWorking;
+        this.showArchived = builder.showArchived;
+        this.showFolders = builder.showFolders;
+        this.showFiles = builder.showFiles;
+        this.sortByDesc = builder.sortByDesc;
+        this.showLinks = builder.showLinks;
+        this.showPages = builder.showPages;
+        this.showDotAssets = builder.showDotAssets;
+        this.languageId = builder.languageId;
+  
+    }
+
+
+    /**
+     * Creates builder to build {@link BrowserQuery}.
+     * 
+     * @return created builder
+     */
+    
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Creates a builder to build {@link BrowserQuery} and initialize it with the given object.
+     * 
+     * @param browserQuery to initialize the builder with
+     * @return created builder
+     */
+    
+    public static Builder from(BrowserQuery browserQuery) {
+        return new Builder(browserQuery);
+    }
+
+    /**
+     * Builder to build {@link BrowserQuery}.
+     */
+    
+    public static final class Builder {
+        private User user ;
+        private String hostFolderId=Folder.SYSTEM_FOLDER;
+        private String filter = null;
+        private List<String> mimeTypes=new ArrayList<>();
+        private List<String>  extensions=new ArrayList<>();
+        private String sortBy="moddate";
+        private int offset=0;
+        private int maxResults=100;
+        private boolean showWorking=true;
+        private boolean showArchived=false;
+        private boolean showFolders=false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using redundant field initializer for 'showFolders'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)